### ISSUE TITLE
Screen not rendering on Intel graphics

### ISSUE THREAD

**Issue by themonglord** [hyprwm/Hyprland#8541]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

Yes

### System Info and Version

Hyprland 0.45.0 built from branch  at commit  dirty ().
Date: 
Tag: , commits: 
built against aquamarine 0.5.0

flags set:
debug

System Information:
System name: Linux
Node name: FreakyArchBTW
Release: 6.11.9-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Sun, 17 Nov 2024 16:06:17 +0000

GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation Skylake GT2 [HD Graphics 520] [8086:1916] (rev 07) (prog-if 00 [VGA controller])

os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo

plugins:
        unknown: not runtime

### Description

This has started happening after I updated hyprland to 0.45.1. Whenever I start hyprland (either via tty or sddm), the screen turns off and stays off. When I switch ttys and go back to the hyprland tty, it shows one frame and it freezes. I am running an Intel I5-6300u with a 520 integrated gpu.

### How to reproduce

Start Hyprland (either via sddm or tty) on the cpu specified on v0.45.1 and above

**Comment by vaxerski** [comment]
can you bisect this? https://wiki.hyprland.org/Crashes-and-Bugs/#bisecting-an-issue

**Comment by themonglord** [comment]
ok, between 0.45.0 and 0.45.1 (where this started happening), it output:
`[c4a77b8da7783d95345935cd2468d83cd54d182b] core: guard pmonitor in focuswindow`

**Comment by vaxerski** [comment]
I don't think you did it correctly if that's the output. Check the git head (latest git) and if it's bad, select that as bad and put 0.45 as good

**Comment by themonglord** [comment]
between 0.45.0 and head
`[83be2480c45bdead9110fdf137a1cd2f5a203378] workspace: fix missing name via focusworkspaceoncurrentmonitor (#8484)
`

**Comment by MightyPlaza** [comment]
are you sure this couldn't be a aquamarine regression?
https://gitlab.archlinux.org/archlinux/packaging/packages/hyprland/-/commit/a6f593168059910617ec760ff3e49f9fbafdc787 was released a day before 0.45 on arch

**Comment by themonglord** [comment]
it is aquamarine. i installed that version of hyprland and it still isn't rendering. pacman doesn't wanna downgrade aquamarine because it apparently breaks this "libaquamarine" dependency that hyprland uses.

**Comment by m4rch3n1ng** [comment]
what version of mesa are you using @themonglord? i noticed the same, also on intel, when upgrading mesa to version `24.3.0` (and reported it to upstream [mesa](https://gitlab.freedesktop.org/mesa/mesa/-/issues/12195)), and the fix for me was downgrading mesa to 24.2.7 (by, in my case, disabling the arch testing repos and running `pacman -Syuu`)

**Comment by vaxerski** [comment]
unless mesa can point to a culprit being hyprland doing something wrong, I disagree.
